extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -271,13 +270,37 @@ public JavaScriptRequestChecker getJavaScriptRequestChecker() {
     }
 
     private class DefaultStaticTenantResolver implements TenantResolver {
+        private static final String DEFAULT_TENANT = "DefaultStaticTenantResolver#DefaultTenant";
```

Comment:
> This resolver is dealing with the convention based path check only, when the tenant id is automatically deduced from the request path so I'd keep it intact.

Alright, sorry, I didn't know. But it is deeply strange convention, sorry, I didn't know about that.

1. What convention speaks about our tenant id stored on `RoutingContext`?
2. I can't see anyone in real-world to name their paths so that they can have tenant name in there. You should name your paths so that they are friendly, user readable paths that makes it clear what the path is about. There can be **false positives** when some paths ends with tenant name. I can't imagine telling users _"do not name your subpaths like tenants just in case we match it for you with some feature you didn't ask for..."_ Anyway, unlike you, I know nothing about this...

Having path-matched tenant with these configuration path-patterns I am proposing is much more useful then matching "ends with" path.

> So, IMHO, the new path-based check should be explicitly dealt with in that same section. It should probably become a 2nd check in that order, straight after a TenantResolver check but before the convention based default resolver check

Sorry, I need to use different words @sberyozkin just to be user I get you right. You propose user-provided `TenantResolver` has priority, after that path-matching (added here) goes and after it, annotation and _ends with_ goes. 

^^^ If I understand it right, sure, I agree. Thanks

Reply:
@michalvavrik Sure, indeed DefaultStaticTenantResolver also checks the tenant id attribute, which makes the effective order:
1. TenantResolver 
2. Tenant annotation which will set that attribute
3. DefaultStaticTenantResolver

(please cleanup my code a bit there and move the tenant attribute check just above the default static resolver check, and remove it from the static resolver)

So the explicitly configured paths should come before the default DefaultStaticTenantResolver check. I'd just like to keep it dedicated to the convention/default check only, instead of mixing it with the other checks, if you'd like you can add `PathMatchingTenantResolver` or similarly named alongside it.

Reply:
Sounds good @sberyozkin.

Reply:
Should be done.